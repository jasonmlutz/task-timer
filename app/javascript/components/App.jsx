import React from 'react';
import { useRoutes } from 'react-router-dom';

import NewUserForm from './Users/NewUserForm';

function App() {
  const routes = useRoutes([
    { path: '/', element: <NewUserForm /> },
  ]);
  return (
    <div>{routes}</div>
  );
}

export default App;
