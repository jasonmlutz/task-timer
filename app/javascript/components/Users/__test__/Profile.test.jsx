/**
 * @jest-environment jsdom
 */
// REACT
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
// TEST UTILITIES
import renderer from 'react-test-renderer';

// COMPONENTS
import App from '../../App';

describe('Profile component', () => {
  describe('STATIC TESTS', () => {
    test('renders correctly from snapshot', () => {
      const tree = renderer.create(
        <MemoryRouter initialEntries={['/user/11235813']}>
          <App />
        </MemoryRouter>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
