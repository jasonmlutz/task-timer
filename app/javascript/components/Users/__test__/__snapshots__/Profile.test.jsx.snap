// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component STATIC TESTS renders correctly from snapshot 1`] = `
<div>
  <div>
    Profile for user id:
    11235813
  </div>
</div>
`;
